src/constants.js

```diff
@@ -6,7 +6,10 @@ export const SC_ATTR = 'data-styled-components'
 export const SC_STREAM_ATTR = 'data-styled-streamed'
 export const CONTEXT_KEY = '__styled-components-stylesheet__'
 
-export const IS_BROWSER = typeof window !== 'undefined'
+export const IS_BROWSER =
+  typeof window !== 'undefined' ||
+  // eslint-disable-next-line no-new-func
+  new Function('try {return this===window;}catch(e){ return false;}')()
```

Comment:
Sorry it’s me again ;) so we have to get rid of Function and the OR.

With the OR the same false positives will occur since the second check won’t be applied when the first succeeds.

Reply:
Damn, you're right. I had tested this in my application, but maybe it used cached files or somehow not picking up the changes.

Many good suggestions in the commit thread, I'll test them.